algoliasearch-common/src/main/java/com/algolia/search/APIClient.java

```diff
@@ -810,13 +811,13 @@ TaskSingleIndex partialUpdateObjects(String indexName, List<Object> objects) thr
   @SuppressWarnings("unchecked")
   <T> BrowseResult<T> browse(String indexName, Query query, String cursor, Class<T> klass) throws AlgoliaException {
     AlgoliaRequest<BrowseResult> algoliaRequest = new AlgoliaRequest<>(
-      HttpMethod.GET,
+      HttpMethod.POST,
       true,
       Arrays.asList("1", "indexes", indexName, "browse"),
       BrowseResult.class,
       klass
     )
-      .setParameters(query.setCursor(cursor).toQueryParam());
+      .setData(query.setCursor(cursor));
```

Comment:
It simplifies the handling of the object `Query` that is needed for the query rules